src/Action/StatusAction.php

```diff
@@ -39,41 +43,57 @@ public function execute($request)
         switch ($quickpayPayment->getState()) {
             case QuickPayPayment::STATE_INITIAL:
                 $request->markNew();
+
                 break;
             case QuickPayPayment::STATE_NEW:
-                $latestOperation = $quickpayPayment->getLatestOperation();
-                if ($latestOperation && $latestOperation->getType() == QuickPayPaymentOperation::TYPE_AUTHORIZE && $latestOperation->isApproved()) {
+                $latestOperation = $this->getLatestOperation($quickpayPayment);
```

Comment:
Before the `$request->markFailed();` would have been called if `$quickpayPayment->getLatestOperation()` returned `null`. Now it will throw an exception if `$quickpayPayment->getLatestOperation()` returns `null`. Is this correct?

Reply:
It does not seem like the places that call for the status can handle exceptions. This will need to be handled in the class.